utils/tinystr/src/ascii.rs

```diff
@@ -22,6 +22,37 @@ impl<const N: usize> TinyAsciiStr<N> {
         Self::from_bytes_inner(bytes, 0, bytes.len(), false)
     }
 
+    /// Creates a `TinyAsciiStr<N>` from a byte slice, replacing invalid bytes.
+    ///
+    /// Null and non-ASCII bytes (i.e. those outside the range `0x01..=0x7F`)
+    /// will be replaced with `0x20`, the space character.
+    ///
+    /// The input slice will be truncated if its length exceeds `N`.
+    pub const fn from_bytes_lossy(bytes: &[u8]) -> Self {
+        const SPACE: u8 = 0x20;
+        let mut out = [0; N];
+        let mut i = 0;
+        // Ord is not available in const, so no `.min(N)`
+        let len = if bytes.len() > N { N } else { bytes.len() };
+
+        // Indexing is protected by the len check above
+        #[allow(clippy::indexing_slicing)]
+        while i < len {
+            let b = bytes[i];
+            if b > 0 && b < 0x80 {
+                out[i] = b;
+            } else {
+                out[i] = SPACE;
```

Comment:
suggestion: traditionally, failed encoding conversions use `?` as the replacement character. "m?me" is more recognizable as an encoding issue than "m me". Another advantage is that it doesn't mess with the naive word count (`.split(' ')`).

Reply:
+1 for `?`

Reply:
Thanks, I agree that's a better choice.